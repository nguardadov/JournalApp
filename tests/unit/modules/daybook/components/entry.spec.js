import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry.vue'
import { journalState } from './../../../mock-data/test-journal-state'
describe('Pruebas en Entry Component', () => {
  //mock router

  const mockRouter = {
    push: jest.fn(),
  }

  const wrapper = shallowMount(Entry, {
    props: {
      entry: journalState.entries[0],
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  })

  test('degbe de hacer match con el snapshot', () => {
    //console.log(wrapper.html())
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de redireccionar al hacer click en el entry-container', () => {
    wrapper.find('.entry-container').trigger('click')
    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toBeCalledWith({
      name: 'entry',
      params: { id: journalState.entries[0].id },
    })
  })

  test('Pruebas en las propiedades computadas', () => {
    //console.log(wrapper.vm.day, wrapper.vm.month, wrapper.vm.yearDay)
    expect(wrapper.vm.day).toBe(3)
    expect(wrapper.vm.month).toBe('Octubre')
    expect(wrapper.vm.yearDay).toBe('2021, Domingo')
  })
})
