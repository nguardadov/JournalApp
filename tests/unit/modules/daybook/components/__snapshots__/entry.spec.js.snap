// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry Component degbe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!--titulo-->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">3</span><span class="mx-1 fs-5">Octubre</span><span class="mx-2 fw-light">2021, Domingo</span></div>
  <div class="entry-description">comienzan los vientos de octubre o vientos de cambios</div>
</div>
`;
